src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputFormattedIncorrectJavadocParagraph.java

```diff
@@ -115,6 +115,7 @@ boolean emulated() {
    *
    * <h1>Testing...</h1>
    */
+  // violation 2 lines above 'block-level tag '<h1>' is preceded by <p> tag'
```

Comment:
You suggested to put violation on block-tag itself in issue description. IMO this is more convenient, it will help user to easily understand the error message and how they can solve it. So we should stick with it. Example from issue description:

```java
  /**
   *  Some summary.
   *
   * <p>
   *  <table> // should give violation: extra P tag. One more possible: after P should be no space
   *  </table>
   * </p>
   */
  public void fooo() {}
```

violation message for context:

```
block-level tag ''<{0}>'' is preceded by <p> tag, preceding <p> tag should be removed.
```

Reply:
it was mistake, please target `p` tag.
We should put violation on target of Check, target is "P" tag.
I will update issue description.

Reply:
issue is updated.

Reply:
We should give violation on the line with the problem. The aim is to fix that problem. If you give violation on a different line, it is not clear what to fix. Users skim messages, it is more mental load to understand that you need to go to a different line to fix it. I'm good with Roman's update.

Reply:
made changes according to Roman's suggestions. New violation message:

https://github.com/checkstyle/checkstyle/blob/03f5d5ecffd7969cf64ab60ab12811ee8ce3d1d0/src/main/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/messages.properties#L23